csharp/src/sdk/Elements/Space.cs

```diff
@@ -0,0 +1,163 @@
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<Line> VerticalEdges()
+        {
+            foreach(var f in Faces())
+            {
+                yield return f.Segments().ElementAt(1);
+            }
+            
+        }
+
+        /// <summary>
+        /// A collection of curves representing the horizontal edges of the space.
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<Line> HorizontalEdges()
+        {
+            foreach(var f in Faces())
+            {
+                yield return f.Segments().ElementAt(0);
+                yield return f.Segments().ElementAt(2);
+            }
+        }
+
+        /// <summary>
+        /// A collection of polylines representing the perimeter of each face of the space.
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<Polyline> Faces()
+        {
+            var b = this.Location.Vertices.ToArray();
+            var t = this.Location.Vertices.ToArray();
+
+            for (var i = 0; i < b.Length; i++)
+            {
+                var next = i + 1;
+                if (i == b.Length - 1)
+                {
+                    next = 0;
+                }
+                var v1 = b[i];
+                var v2 = b[next];
+                var v3 = t[next];
+                var v4 = t[i];
+                var v1n = new Vector3(v1.X, v1.Y, this.BottomElevation);
+                var v2n = new Vector3(v2.X, v2.Y, this.BottomElevation);
+                var v3n = new Vector3(v3.X, v3.Y, this.BottomElevation + this.Height);
+                var v4n = new Vector3(v4.X, v4.Y, this.BottomElevation + this.Height);
+                yield return new Polyline(new[] { v1n, v2n, v3n, v4n });
+            }
+        }
+
+        /// <summary>
+        /// For each face of the space apply a creator function.
+        /// </summary>
+        /// <param name="creator">The function to apply.</param>
+        /// <typeparam name="T">The creator function's return type.</typeparam>
+        /// <returns></returns>
+        public IEnumerable<T> ForEachFaceCreateElementsOfType<T>(Func<Polyline, IEnumerable<T>> creator)
+        {
+            var results = new List<T>();
+            foreach(var p in Faces())
+            {
+                results.AddRange(creator(p));
+            }
+            return results;
+        }
+
+       
+        /// <summary>
+        /// Tessellate the mass.
+        /// </summary>
+        /// <returns>A mesh representing the tessellated mass.</returns>
+        public Mesh Tessellate()
+        {
+            var mesh = new Mesh();
+            foreach (var s in Faces())
+            {
+                mesh.AddQuad(s.ToArray());
+            }
+
+            mesh.AddTesselatedFace(new[] { this.Location }, this.BottomElevation);
+            mesh.AddTesselatedFace(new[] { this.Location }, this.BottomElevation + this.Height, true);
```

Comment:
Following on the comment above, this would just be `mesh.AddTesselatedFace(new[] { this.Location }, this.Height, true);`

Reply:
Right.

Reply:
Used the way Floor handles it instead.